### PR TITLE
Merge CONTRIBUTING.rst into docs/hacking/

### PR DESCRIPTION
**Pull request by seanh**
This gets all our docs about contributing to the project in one place,
and removes some duplication.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1,172 +1,4 @@
-***************************
-Contributing to development
-***************************
+For contributing h (whether code, bug reports, translations, documentation,
```
Comment (nickstenning) [inline CONTRIBUTING.rst]: s/contributing/contributing to/

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -1,172 +1,4 @@
...
+etc.) see our contributing guidelines:
```
Comment (nickstenning) [inline CONTRIBUTING.rst]: Any reason you wouldn't use an RST link here?

**Comment by seanh** [comment]
This is again just reorg and dedup, very little actual editing. I did actually remove this though:

> Committer policy
> 
> "Committers" are those with push access to the main repository. These people should feel free to commit small changes in good faith. It is expected that these people should read upstream commits made by others when they feel qualified to review the material and comment with any objections, questions or suggestions. In general, these commits should be uncontroversial and do not require up front code review.
> 
> Larger changes, and changes being submitted by non-committers, should follow the branching and merging strategy outlined in the next section.

It just didn't seem like something we needed in the docs

**Comment by seanh** [comment]
Cripes

File: CONTRIBUTING.rst (hunk 2 above)
Comment (tilgovi) [inline CONTRIBUTING.rst]: I'm clearly moving too fast :). And I haven't even had coffee yet today.

Comment (seanh) [inline CONTRIBUTING.rst]: Don't think so

**Pull request merged**